### PR TITLE
feat: user post notification worker

### PR DESCRIPTION
**Pull request by capJavert**
AS-586

### CODE REVIEW

File: src/workers/newNotificationV2Mail.ts [hunk 1]
```diff
@@ -681,9 +682,30 @@ const notificationToTemplateData: Record<NotificationType, TemplateDataFunc> = {
...
-  user_post_added: async () => {
-    return null;
+  user_post_added: async (con, user, notification) => {
+    const [post, avatar] = await Promise.all([
+      await con.getRepository(Post).findOneOrFail({
+        where: {
+          id: notification.referenceId,
+        },
+      }),
+      con.getRepository(NotificationAvatarV2).findOneOrFail({
```
Comment (omBratteng) [inline src/workers/newNotificationV2Mail.ts:692]: one is `await`, one is not?

File: src/workers/newNotificationV2Mail.ts [hunk 2]
```diff
@@ -80,7 +81,7 @@ export const notificationToTemplateId: Record<NotificationType, string> = {
   streak_reset_restore: '',
   squad_featured: '56',
   // TODO AS-534 add template id
-  user_post_added: '',
+  user_post_added: 'test',
```
Comment (rebelchris) [inline src/workers/newNotificationV2Mail.ts:84]: Just blocking for now

File: src/workers/newNotificationV2Mail.ts [hunk 3]
```diff
@@ -682,8 +683,30 @@ const notificationToTemplateData: Record<NotificationType, TemplateDataFunc> = {
     };
   },
   // TODO AS-534 add handler
```
Comment (rebelchris) [inline src/workers/newNotificationV2Mail.ts]: TODO removal

**Review by omBratteng — APPROVED** [review]
LGTM, just one minor comment

File: src/workers/notifications/postAddedUserNotification.ts [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+        userPromises.push(postEntity.scout);
+      }
+
+      if (userPromises.length === 0) {
+        return;
+      }
+
+      const users = await Promise.all(userPromises);
+
+      const contentPreferenceQuery = con
```
Comment (rebelchris) [inline src/workers/notifications/postAddedUserNotification.ts:49]: Probably have to monitor if this query becomes heavy on PG

**Review by rebelchris — CHANGES_REQUESTED** [review]
Blocking on template // todo

File: src/workers/newNotificationV2Mail.ts (hunk 2 above)
Comment (capJavert) [reply src/workers/newNotificationV2Mail.ts:84]: sorted

**Review by rebelchris — APPROVED** [review]

**Pull request merged**